src/modules/billing/services/two-part-tariff-service/index.js

```diff
@@ -0,0 +1,91 @@
+const { set } = require('lodash');
+const twoPartTariffMatching = require('./two-part-tariff-matching');
+const returnHelpers = require('./returns-helpers');
+const Purpose = require('../../../../lib/models/purpose');
+
+const mapResultsWithData = (overallError, data, transactions) => {
```

Comment:
Because the transactions passed in are not cloned, I think these functions are mutating the transactions passed in. 
In this case it should perhaps be named `decorateTransactions` or similar.